Dockerfile

```diff
@@ -11,18 +14,23 @@ COPY go.sum go.sum
 RUN go mod download
 
 # Copy the go source
-COPY main.go main.go
-COPY controllers/ controllers/
+COPY cmd/main.go cmd/main.go
+COPY internal/controller/ internal/controller/
 COPY pkg/ pkg/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
```

Comment:
shouldn't goarch be GOARCH=${TARGETARCH:-amd64}  ?

Reply:
No, because it will not work by default on other arches. If you omit TARGETARCH, it will take yours.